auditing_standards.md

```diff
@@ -237,3 +237,76 @@ Crates with this criteria contain very dangerous unsafe rust code.
+
+For example, even though
+[the 1.0.79 to 1.0.80 delta of the `proc-macro2` crate](https://chromium-review.googlesource.com/c/chromium/src/+/5453972/2..6)
+doesn't add or change any `unsafe` code, it shouldn't be recorded as
+`ub-risk-0`, because there is still `unsafe` outside of the delta
+(e.g. `unsafe { imp::Literal::from_str_unchecked(repr) }` in
+[`lib.rs` on line 1282](https://chromium-review.googlesource.com/c/chromium/src/+/5453972/2..6/third_party/rust/chromium_crates_io/vendor/proc-macro2-1.0.80/src/lib.rs#1282)).
+
+### Audits should ignore criteria required by `config.toml` policy
+
+Audits should record the actual risk of the crate, even if a looser
+certification would be sufficient to pass a `config.toml` policy.
+
+For example, Chromium requires `ub-risk-2` for crates used in the browser
+process - e.g. see the following auto-generated `config.toml`
+[entry for the `quote` crate](https://source.chromium.org/chromium/chromium/src/+/main:third_party/rust/chromium_crates_io/supply-chain/config.toml;l=158-159;drc=aaf3f1dab40ed4248391227583a1590bfbf7062e):
+
+```
+# supply-chain/config.toml:
+[policy."quote:1.0.36"]
+criteria = ["does-not-implement-crypto", "safe-to-deploy", "ub-risk-2"]
+```
+
+This means that certifying this crate as `ub-risk-2` is sufficient for `cargo
+vet` checks/presubmits to pass and `cargo vet check` will actually only ask
+to certify at the `ub-risk-2` level - see the example below:
+
+```
+$ tools/crates/run_cargo_vet.py check
+Vetting Failed!
+
+1 unvetted dependencies:
+  quote:1.0.36 missing ["safe-to-deploy", "does-not-implement-crypto", "ub-risk-2"]
+
+recommended audits for safe-to-deploy, does-not-implement-crypto, ub-risk-2:
+    Command                             Publisher  Used By                                    Audit Size
+    cargo vet diff quote 1.0.35 1.0.36  dtolnay    syn, chromium, prost-derive, and 6 others  6 files changed, 568 insertions(+), 548 deletions(-)
+
+estimated audit backlog: 1116 lines
+
+Use |cargo vet certify| to record the audits.
+```
+
+Nevertheless, the audit should ignore the policy requirements and certify
+`quote` 1.0.36 as `ub-risk-0` because it doesn't have any `unsafe` code.
+
+### Conditionally-compiled code: TODO
+
+Some crates may contain code that may be excluded in some build environments.
+Some non-exhaustive examples:
+
+* Code controlled by crate features (e.g. `#[cfg(feature = "foo")]`
+* Code applicable only to certain targets (e.g. `#[cfg(target_arch = "powerpc")]`
+  or `#[cfg(target_os = "solaris")]`)
+
+**TODO**: Spell out the guidelines for auditing such conditional code.
+Consider that:
+
+* It is impossible to predict what code may be enabled by consumers of `audits.toml`.
+  For example, Chromium may rely on `audits.toml` imported from Fuchsia, even
+  though Chromium targets additional operating systems.
+* `cargo vet` has no way to indicate partial audits - see:
+  https://github.com/mozilla/cargo-vet/issues/380
```

Comment:
Can we move this to a separate PR since we don't have a consensus yet?

Reply:
Done